public/app/features/alerting/unified/Silences.test.tsx

```diff
@@ -0,0 +1,260 @@
+
+    expect(ui.editor.matcherName.query(matchers[2])).toHaveValue('hello');
+    expect(ui.editor.matcherOperator(MatcherOperator.notEqual).query(matchers[2])).not.toBeNull();
+    expect(ui.editor.matcherValue.query(matchers[2])).toHaveValue('world');
+
+    expect(ui.editor.matcherName.query(matchers[3])).toHaveValue('cluster');
+    expect(ui.editor.matcherOperator(MatcherOperator.notRegex).query(matchers[3])).not.toBeNull();
+    expect(ui.editor.matcherValue.query(matchers[3])).toHaveValue('us-central.*');
+  });
+
+  it('creates a new silence', async () => {
+    renderSilences(baseUrlPath);
+    await waitFor(() => expect(ui.editor.durationField.query()).not.toBeNull());
+
+    const start = new Date();
+    const end = new Date(start.getTime() + 60 * 60 * 1000);
+
+    userEvent.click(ui.editor.timeRange.get());
+    userEvent.clear(ui.editor.timeRangeStart.get());
+    await userEvent.type(
+      ui.editor.timeRangeStart.get(),
+      `${start.getFullYear()}-${
+        start.getMonth() + 1
+      }-${start.getDate()} ${start.getHours()}:${start.getMinutes()}:${start.getSeconds()}`
+    );
+
+    userEvent.clear(ui.editor.timeRangeEnd.get());
+    await userEvent.type(
+      ui.editor.timeRangeEnd.get(),
+      `${end.getFullYear()}-${
+        end.getMonth() + 1
+      }-${end.getDate()} ${end.getHours()}:${end.getMinutes()}:${end.getSeconds()}`
+    );
+    userEvent.click(ui.editor.applyTimeRange.get());
+
+    await waitFor(() => expect(ui.editor.durationInput.query()).toHaveValue('1h'));
+
+    await userEvent.type(ui.editor.matcherName.get(), 'foo');
+    await userEvent.type(ui.editor.matcherOperatorSelect.get(), '=');
+    userEvent.tab();
+    await userEvent.type(ui.editor.matcherValue.get(), 'bar');
+
+    userEvent.click(ui.editor.addMatcherButton.get());
+    await userEvent.type(ui.editor.matcherName.getAll()[1], 'bar');
+    await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[1], '!=');
+    userEvent.tab();
+    await userEvent.type(ui.editor.matcherValue.getAll()[1], 'buzz');
+
+    userEvent.click(ui.editor.addMatcherButton.get());
+    await userEvent.type(ui.editor.matcherName.getAll()[2], 'region');
+    await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[2], '=~');
+    userEvent.tab();
+    await userEvent.type(ui.editor.matcherValue.getAll()[2], 'us-west-.*');
+
+    userEvent.click(ui.editor.addMatcherButton.get());
+    await userEvent.type(ui.editor.matcherName.getAll()[3], 'env');
+    await userEvent.type(ui.editor.matcherOperatorSelect.getAll()[3], '!~');
+    userEvent.tab();
+    await userEvent.type(ui.editor.matcherValue.getAll()[3], 'dev|staging');
+
+    await userEvent.type(ui.editor.comment.get(), 'Test');
+    await userEvent.type(ui.editor.createdBy.get(), 'Homer Simpson');
+
+    userEvent.click(ui.editor.submit.get());
+
+    await waitFor(() =>
+      expect(mocks.api.createOrUpdateSilence).toHaveBeenCalledWith(
+        'grafana',
+        expect.objectContaining({
```

Comment:
:open_mouth:  niiice, didn't know this can be done!

Reply:
Yeah I just found out about this. The timestamps for `startsAt` and `endsAt` were giving me such a headache